ISISScientificComputing/autoreduce
# Implement archive monitor to poll data archive directory

**Description of changes**
* Implemented the archive monitor to poll the ISIS data archive for files that may have been missed by watchdog observing EndOfRunMonitor.  
* Added generated data to the mock database being used for testing
* Added a data archive creator to create a fake data archive that can be used for testing
* Refactored monitors to use generic windows service class instead of duplicating code


**How to test**

* Ensure the unit tests pass and travis is happy. 

**Dev if data is coming through**

* Deploy this on development and turn off the EndOfRunMonitor. 
  * We may need to wait some time before a run is found, but this monitor should pick up the difference between the reduction database and the archive and run resubmit the job. 

**Locally**

* It should be possible to test this by:
  * [creating a local archive](https://github.com/ISISScientificComputing/autoreduce/wiki/Testing-and-validating#creating-a-fake-data-archive)
  * Set the `utils/settings.py` `INST_PATH` to be the file directory you created the archive in e.g. if I made the archive in `C:\data-archive` it would be: `INST_PATH = os.path.join('C:\\', 'data-archive', 'NDX{}', 'Instrument')`
  * Install the archive monitor service with `python isis_archive_monitor_win_service.py install`
  * You may wish to lower the polling rate (by default it is 10mins) you can do this in the `isis_archive_monitor_helper.py`
  * Start the service running in the service manager
  * Add data files to the archive data directories and check the logs pick them up correctly. If you like you can hook these into the dev node and check they get pushed through to the database/webapp

Fixes #11
